### PR TITLE
Updated load balancer docs in DPG; HA guide 

### PR DESCRIPTION
**Pull request by dayleparker**
Added instructions to the load balancer section in the Deployment Planning Guide for extra config; added a briefer version to the HAProxy config in the HA guide, to align content better.

I've also added links to the Gen Config guide (has instructions on editing the Advanced settings), and to the reference arch for Deploying CloudForms at Scale (just for downstream), as it also provides these instructions, plus cross-linked these two guides for a better user experience.

Any comments welcome -- I think the distinction of this config being for "user interface appliances" or "UI appliances" is important here, despite the fact that we generally don't use this term in our docs. Time to re-evaluate that perhaps?

### CODE REVIEW

**Comment by dayleparker** [comment]
Hi Chris,

Many thanks for volunteering to review this one -- please let me know your thoughts on the terminology (and how it aligns with similar topics you've worked on), the flow, etc. I'm still a bit undecided also on whether the addition to the HA guide should be in its own section, and single-sourced from our side, or if it's OK in an admonition. What do you think?

Any feedback welcome :)
Cheers,
Dayle

File: doc-Deployment_Planning_Guide/topics/planning.adoc [hunk 1]
```diff
@@ -212,9 +212,48 @@ Providers can be added at any level of tenancy. Once added, a provider is visibl
...
+.Load balancer configuration
+
+* Configure the load balancer to use sticky sessions. This ensures that when a session is started, all requests for that session are sent to the same worker appliance.
+* Configure the load balancer to test for connectivity using the {product-title_short} ping response page: `https://appliance_name/ping`. The expected reply from the appliance is the text string _pong_. Using this URL is preferable to the appliance login URL as it does not establish a connection to the database. 
+
+.Appliance configuration
+
+By default, {product-title_short} user interface worker appliances store session data in the local appliance’s memcache. When operating behind a load balancer, it is better to store the user session data in the shared database (VMDB) rather than in memcache, which is not shared.
```
Comment (cbudz) [inline doc-Deployment_Planning_Guide/topics/planning.adoc]: I would cut the first two sentences here. The following section more directly tells the user what they need to do when operating UI appliances behind a load balancer.

File: doc-Deployment_Planning_Guide/topics/planning.adoc [hunk 2]
```diff
@@ -212,9 +212,48 @@ Providers can be added at any level of tenancy. Once added, a provider is visibl
...
+.Load balancer configuration
+
+* Configure the load balancer to use sticky sessions. This ensures that when a session is started, all requests for that session are sent to the same worker appliance.
+* Configure the load balancer to test for connectivity using the {product-title_short} ping response page: `https://appliance_name/ping`. The expected reply from the appliance is the text string _pong_. Using this URL is preferable to the appliance login URL as it does not establish a connection to the database. 
+
+.Appliance configuration
+
+By default, {product-title_short} user interface worker appliances store session data in the local appliance’s memcache. When operating behind a load balancer, it is better to store the user session data in the shared database (VMDB) rather than in memcache, which is not shared. 
+
+When using a load balacner, configure appliances that have the *User Interface* role enabled to store session data in the database. As a result, the user does not need to re-login if the load balancer redirects them to an alternative server in case the original user interface worker is unresponsive.
```
Comment (cbudz) [inline doc-Deployment_Planning_Guide/topics/planning.adoc]: If you remove the first section, I'd just revide 'database' to something like "shared database (VMDB)"

File: doc-Deployment_Planning_Guide/topics/planning.adoc [hunk 3]
```diff
@@ -212,9 +212,48 @@ Providers can be added at any level of tenancy. Once added, a provider is visibl
...
++
+----
+:server:
+...
+ :session_store: sql
+----
++
+. Click *Save*. 
+
+Configure the `session_store` parameter to point to `sql` on each user interface appliance behind the load balancer.
```
Comment (cbudz) [inline doc-Deployment_Planning_Guide/topics/planning.adoc]: This detail seems like it should be elevated and possibly made into a note or important admonition.

File: doc-Deployment_Planning_Guide/topics/planning.adoc [hunk 4]
```diff
@@ -212,9 +212,48 @@ Providers can be added at any level of tenancy. Once added, a provider is visibl
...
+Also see https://access.redhat.com/documentation/en-us/reference_architectures/2017/html-single/deploying_cloudforms_at_scale/#load_balancers[Load Balancers] in the _Deploying CloudForms at Scale_ reference architecture for further information. 
+
+For information on configuring database failover in VMDB appliances, see https://access.redhat.com/documentation/en-us/red_hat_cloudforms/4.6/html-single/high_availability_guide/[the _High Availability Guide_].
+endif::cfme[]
+
+
+[[load-balancer-ssui-addl-config]]
+==== Additional Configuration for Load Balancers and the Self Service User Interface
+
+Running several appliances behind a load balancer can cause login issues from the Self Service user interface because `memcached`, the service storing the session in memory, is not shared between the appliances. To prevent this, configure your environment using the steps below to run the `memcached` service on one appliance and have it shared with the other appliances.
```
Comment (cbudz) [inline doc-Deployment_Planning_Guide/topics/planning.adoc]: Are we talking about appliances in general or is this still aligned with UI appliances behind a load balancer? I would read this as still referencing UI appliances, due to continuity from the above sections.

**Comment by cbudz** [comment]
I think the distinction of this config being for "user interface appliances" or "UI appliances" is important here, despite the fact that we generally don't use this term in our docs. Time to re-evaluate that perhaps?

- I'm torn here. Part of me wants to say, "yes, let's re-evaluate" but I think this is a term more applicable to the reference architecture docs than anywhere else. I don't think it would hurt to create a product specifc convention highlighting that use of "UI appliace" refers specifically to the appliance(s) in the system architecture dedicated to the user inferface server role. 

...please let me know your thoughts on the terminology 
-  All good here. I think anyone going into this configuration exercise will be familiar with the terminology. 

..HA Proxy:

- I think the IMPORTANT admonition works well, since its a critical topic about which the admin should be aware. 

I hope these notes help. Let me know if you have any further questions. 

-Chris

File: doc-Deployment_Planning_Guide/topics/planning.adoc (hunk 3 above)
Comment (dayleparker) [reply doc-Deployment_Planning_Guide/topics/planning.adoc]: I've made it "Important", as the config won't work as described without configuring this on each UI appliance. I also mentioned this at the beginning of the procedure now. Thanks for highlighting that.

File: doc-Deployment_Planning_Guide/topics/planning.adoc (hunk 4 above)
Comment (dayleparker) [reply doc-Deployment_Planning_Guide/topics/planning.adoc]: Correct. I've edited the start of this section now to use "Running several user interface worker appliances behind a load balancer" (clunky but clearer anyway)

**Comment by dayleparker** [comment]
Hi Chris,

Thanks so much for your helpful review and comments. I've fixed the things you highlighted, and a few other things I noticed when going through this section again, notably:
- I had linked to the 4.5 Gen Config guide, changed to 4.6.
- Added a sub-section structure which I hope makes topics clearer in the table of contents:
lb-toc

- I realized the instructions for configuring session_store to `sql` pretty much directly conflicted with the ones in the load balancer/SSUI  section, so I've made a clarification in both parts which i hope (!) tell people when which config is better. I don't really know how common it is to use a LB with the SSUI or without the SSUI, so I'll seek feedback in the bug about that (which could explain some of the confusion in the comments). 

Let me know what you think, and if you spot any more changed needed :)

**Comment by dayleparker** [comment]
Hey @cbudz,
Thanks for reviewing again, and apologies for the hold-up. Thanks to Nick's info in the BZ, I can now remove the SSUI section for master/4.6/4.5, which makes it way less confusing (and no more long section title!). 

I've done this in master and don't anticipate more changes; if it looks OK to you, please feel free to merge and I will commence cherry-picking.
Thanks so much again!
Dayle

**Comment by cbudz** [comment]
Hey @dayleparker 
All looks good, I'm merging this PR.

-Chris

**Pull request merged**